doc/source/train/user_guide.rst

```diff
@@ -498,10 +514,30 @@ useful for:
 3. Persisting trained model state to later use for serving/inference.
 4. In general, storing any model artifacts.
 
+Reporting intermediate results and handling checkpoints
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ray AIR provides a :ref:`Session <air-session-key-concepts>` API for reporting intermediate
+results and checkpoints from the training function (run on distributed workers) up to the
+``Trainer`` (where your python script is executed) by calling ``session.report(metrics)``.
+
+The primary use-case for reporting is for metrics (accuracy, loss, etc.) at
+the end of each training epoch.
+
+.. code-block:: python
+
+    from ray.air import session
+
+    def train_func():
+        ...
+        for i in range(num_epochs):
+            results = model.train(...)
+            session.report({"results": results})
```

Comment:
I feel like this is missing some important details around what happens to these reported metrics.

(Second to this, maybe we should mention that only worker 0 results are used).